app/com/m3/octoparts/model/config/HttpProxySettings.scala

```diff
@@ -0,0 +1,52 @@
+package com.m3.octoparts.model.config
+
+import com.netaporter.uri.config.UriConfig
+import com.netaporter.uri.parsing.DefaultUriParser
+import org.parboiled2._
+
+import scala.util.Try
+
+case class HttpProxySettings(scheme: String, host: String, port: Int) {
+
+  import com.m3.octoparts.model.config.HttpProxySettings._
+
+  def serialize: String = {
+    val schemePart = if (scheme == DefaultScheme) "" else s"$scheme://"
```

Comment:
Convention, like in most browsers' settings page (by default the proxy operates over HTTP)